endpoint.go

```diff
@@ -600,7 +606,23 @@ func (ep *endpoint) sbLeave(sbox Sandbox, options ...EndpointOption) error {
 	return sb.clearDefaultGW()
 }
 
-func (ep *endpoint) Delete() error {
+func (n *network) validateForceDelete(locator string) error {
+	if n.Scope() == datastore.LocalScope {
+		return nil
+	}
+
+	if locator == "" {
+		return fmt.Errorf("invalid endpoint locator identifier")
+	}
+
+	if n.getController().isNodeAlive(locator) {
```

Comment:
@vieux based on the design-review comments from @mrjana we decided to add this check to prevent the `-f` from cleaning up the endpoint when the remote host holding the endpoint is still alive. At the engine level, the node liveness is a factor of discovery timer (by default 20 * 3 sec). This means, swarm cannot push a force delete till the engine catches up with the node failure.
We need this check in order for protecting the engine from invalid forced delete which can lead to inconsistent states and the user can get confused.